### ISSUE TITLE
Replace uses of deprecated rules_go file embedding 

### ISSUE THREAD

**Issue by prestonvanloon** [prysmaticlabs/prysm#12688]
# 💎 Issue

### Background

Prysm has been facing this warning for some time now.

```
Embedding is now better handled by using rules_go's built-in embedding functionality (https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md#go_library-embedsrcs). The `go_embed_data_dependencies` macro is deprecated and will be removed in rules_go version 0.39.
```

The use of `go_embed_data_dependencies`, and `go_embed_data` has been deprecated for some time and will need to be updated in Prysm soon.

### Description

This is used to embed the prysm web UI here

https://github.com/prysmaticlabs/prysm/blob/58df1f1ba5996d299c1c1d5874464884ee551be8/validator/web/BUILD.bazel#L26-L32

Rather than using go_embed_data, we should be using `//go:embed` with embedsrcs. Review the documentation for go_library https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/rules.md#go_library-embedsrcs. Once go_embed_data is no longer used, we can remove go_embed_data_dependencies() from the WORKSPACE file.

https://github.com/prysmaticlabs/prysm/blob/58df1f1ba5996d299c1c1d5874464884ee551be8/WORKSPACE#L384-L386

**Comment by kagzgenius** [comment]
Hi, I would love to give this a go!

**Comment by rkapka** [comment]
You are assigned now @kagzgenius

**Comment by prestonvanloon** [comment]
@kagzgenius Has any progress been made on this issue? It is required for us to update to the latest version of rules_go. Thanks!

**Comment by prestonvanloon** [comment]
I discussed this offline with @james-prysm, @rkapka, and @rauljordan. The web UI is deprecated and will be removed or replaced in the future, as I mentioned in my previous comment. However, we are not looking to remove it at this time so the solution we have landed on is to update the web UI with any relevant changes one final time and then "freeze" the UI by removing the embedding macro and relying solely on the generated file `validator/web/site_data.go`

For the build file, the diff would look like
```diff
diff --git a/validator/web/BUILD.bazel b/validator/web/BUILD.bazel
index 0bfc3e824..23529e474 100644
--- a/validator/web/BUILD.bazel
+++ b/validator/web/BUILD.bazel
@@ -1,34 +1,20 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_test")
-load("@prysm//tools/go:def.bzl", "go_library")
-load("@io_bazel_rules_go//extras:embed_data.bzl", "go_embed_data")
+load("@prysm//tools/go:def.bzl", "go_library", "go_test")
 
-# gazelle:ignore site_data.go
 go_library(
     name = "go_default_library",
     srcs = [
         "doc.go",
         "handler.go",
-        "log.go",
         "headers.go",
-        ":site_data",  # keep
+        "log.go",
+        "site_data.go",
     ],
     importpath = "github.com/prysmaticlabs/prysm/v4/validator/web",
     visibility = [
         "//api/gateway:__pkg__",
         "//validator:__subpackages__",
     ],
-    deps = [
-        "//runtime:go_default_library",
-        "@com_github_sirupsen_logrus//:go_default_library",
-    ],
-)
-
-go_embed_data(
-    name = "site_data",
-    srcs = [
-        "@prysm_web_ui//:site",
-    ],
-    var = "site",
+    deps = ["@com_github_sirupsen_logrus//:go_default_library"],
 )
 
 go_test(
```

The usage of `@io_bazel_rules_go//extras:embed_data_deps.bzl` in the WORKSPACE file would be removed. 

To summarize:

Pull request 1:
- [ ] Update `validator/web/site_data.go` (@james-prysm)

Pull request 2:
- [ ] Remove `prysm_web_ui` `http_archive` from WORKSPACE
- [ ] Remove `go_embed_data_dependencies()` from WORKSPACE
- [ ] Remove `go_embed_data` target from `validator/web/BUILD.bazel`
- [ ] Remove `# gazelle:ignore site_data.go` annotation from `validator/web/BUILD.bazel`
- [ ] Run `bazel run //:gazelle`

That should do effectively freeze the web UI with the latest changes and remove the use of the go_embed_data macro.

**Comment by kagzgenius** [comment]
Hi @prestonvanloon, thanks for the clarification. I was unsure how to approach this, but this suggestion helps a ton!
